server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/ProcessingContext.scala

```diff
@@ -82,14 +84,14 @@ object JsonPath {
         } else if (arrayElementPartPosition == 0) {
           asArrayElementPart(string)
         } else {
-          asArrayElementInAnObject(string, part, arrayElementPartPosition)
+          asArrayElementInAnObject(part, arrayElementPartPosition)
         }
     })
 
   private def asPlainPart(part: String): List[JsonPathPart] = List(PlainPart(part))
 
-  private def asArrayElementInAnObject(string: String, part: String, arrayElementPartPosition: Int): List[JsonPathPart] =
-    ArrayElementPart.parse(string.substring(arrayElementPartPosition))
+  private def asArrayElementInAnObject(part: String, arrayElementPartPosition: Int): List[JsonPathPart] =
+    ArrayElementPart.parse(part.substring(arrayElementPartPosition))
```

Comment:
I don't sure about this line

Reply:
This change coverage test case `"succeed when first array element is present and specific path"`
```scala
 "succeed when first array element is present and specific path" in {
      val jsonPath = JsonPath.parse("path[0]/id")
      val json = Json.parse("""{"path":[{"id":"1","code":"a"},{"id":"2","code":"b"},{"id":"3","code":"c"}]}""".stripMargin)
      val expected = JsString("1")

      jsonPath.evaluate(json) should equal(JsSuccess(expected))
    }
```

Reply:
I think it is because with the previous syntax a JSON path path (what is between `/`) could be both a property with an array element accessor: 2 in 1. (eg `a/b[1]/c` have a `b[1]` part that is actually `property b` + `element 1`).

The right JSON path syntax is simpler and have a 1 to 1 relationship: `a/b/1/c` : we can likely simplify `JsonPath.parse` a looot!